runtime/gc_glue_java/CompactSchemeFixupObject.cpp

```diff
@@ -50,19 +54,42 @@ MM_CompactSchemeFixupObject::fixupMixedObject(omrobjectptr_t objectPtr)
 }
 
 void
-MM_CompactSchemeFixupObject::doStackSlot(MM_EnvironmentBase *env, omrobjectptr_t fromObject, omrobjectptr_t *slot)
+MM_CompactSchemeFixupObject::doSlot(MM_EnvironmentBase *env, omrobjectptr_t fromObject, omrobjectptr_t *slotPtr)
+{
+	*slotPtr = _compactScheme->getForwardingPtr(*slotPtr);
+}
+
+#if JAVA_SPEC_VERSION >= 24
+void
+MM_CompactSchemeFixupObject::doContinuationSlot(MM_EnvironmentBase *env, omrobjectptr_t fromObject, omrobjectptr_t *slotPtr, GC_ContinuationSlotIterator *continuationSlotIterator)
 {
-	*slot = _compactScheme->getForwardingPtr(*slot);
+	if (isHeapObject(*slotPtr)) {
+		doSlot(env, fromObject, slotPtr);
+	} else if (NULL != *slotPtr) {
+		Assert_MM_true(GC_ContinuationSlotIterator::state_monitor_records == continuationSlotIterator->getState());
+	}
 }
+#endif /* JAVA_SPEC_VERSION >= 24 */
 
+void
+MM_CompactSchemeFixupObject::doStackSlot(MM_EnvironmentBase *env, omrobjectptr_t fromObject, omrobjectptr_t *slotPtr, J9StackWalkState *walkState, const void *stackLocation)
```

Comment:
I believe doStackSlot is just used for continuation stack slot scanning. If so, we should rename it to doContinuationStackSlot.

Here and for any other class. If I'm wrong (if for some other class, it's shared for standard threads as well, let me know).

For the same reason  stackSlotIteratorForCompactScheme should be renamed to continuaitonStackSlotIteratorForCompactScheme

Reply:
Actually, even though continuation is the only user of doStackSlot/stackSlotIteratorForCompactScheme I think it's ok to leave the names generic, since there is nothing in their implemention that is continuation specific. Potentially, they could be used for regular threads, too.